src/test/java/eu/dissco/backend/repository/SpecimenRepositoryIT.java

```diff
@@ -0,0 +1,88 @@
+package eu.dissco.backend.repository;
+
+import static eu.dissco.backend.TestUtils.MAPPER;
+import static eu.dissco.backend.TestUtils.givenDigitalSpecimen;
+import static eu.dissco.backend.database.jooq.Tables.NEW_DIGITAL_SPECIMEN;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import eu.dissco.backend.domain.DigitalSpecimen;
+import java.util.ArrayList;
+import org.jooq.JSONB;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SpecimenRepositoryIT extends BaseRepositoryIT {
+
+  private SpecimenRepository repository;
+
+  @BeforeEach
+  void setup() {
+    repository = new SpecimenRepository(context, MAPPER);
+  }
+
+  @AfterEach
+  void destroy() {
+    context.truncate(NEW_DIGITAL_SPECIMEN).execute();
+  }
+
+  @Test
+  void testGetSpecimenLatest() throws JsonProcessingException {
+    // Given
+    populateSpecimenTable();
+
+    // When
+    var resultPage1 = repository.getSpecimensLatest(1, 15);
+    var resultPage2 = repository.getSpecimensLatest(2, 15);
+
+    // Then
+    assertThat(resultPage1).hasSize(15);
+    assertThat(resultPage2).hasSize(7);
+  }
+
+  @Test
+  void testGetSpecimenById() throws JsonProcessingException {
+    // Given
+    populateSpecimenTable();
+
+    // When
+    var result = repository.getSpecimenById("20.5000.1025/ABC-123-XY3");
+
```

Comment:
Should use an id from testUtils

Reply:
No not in this case. I inserted a bunch of records and to make them unique the get a number at the end. So the TestUtils.ID (20.5000.1025/ABC-123-XY3) won't match any of these.